### ISSUE TITLE
Serve a unique favicon.ico per project

### ISSUE THREAD

**Issue by wmhilton** [jus/jus#62]
Browser's tend to cache the favicon, and so whenever I start a new project and am serving on http://localhost:3000 it will use the icon from some other project I'm working on. What's more, Chrome will add 404 errors for `favicon.ico` to the network request log and the console, and my heart rate goes up whenever Chrome starts printing red text in the console. This is one of those little bitty irksome things that bothers me *just enough* over time that I am resolved that `jus serve` be able to invent a new `favicon.ico` on the fly and serve it for projects that don't have a favicon.

My plan is to use [identicon](https://www.npmjs.com/package/identicon) to generate the favicons, and then make an express middleware that can plugged into jus. [This page](http://www.cs.bme.hu/~dhanak/identfavicon/) shows examples what identicons look like:
![image](http://www.cs.bme.hu/~dhanak/identfavicon/screenshot2.png)

**Comment by wmhilton** [comment]
No code to merge yet, because I like being able to install `jus` and so we can't add 'identicon' as a dependency yet. But when we do, my code looks something like this:

```js
// lib/server.js
const express  = require('express')
const cors     = require('cors')
...
const identicon = require('identicon')
const favicon   = require('serve-favicon')

var server = module.exports = express()
server.use(cors())

// Create unique favicon from the current working directory
var faviconPNG = identicon.generateSync({id: process.cwd(), size: 16})
server.use(favicon(faviconPNG))
...
```

**Comment by jdormit** [comment]
Looks good, but is there a check to serve an existing favicon if one exists?

**Comment by wmhilton** [comment]
Right, obviously an actual favicon.ico would take priority. I suppose that would mean putting `server.use(favicon(faviconPNG))` farther down, as a fallback for the static server.

**Comment by zeke** [comment]
So this is just for development? Or do you also intend to kick out a default favicon in the build process if none exists?

**Comment by wmhilton** [comment]
I was just thinking of during development, to help me keep my tabs straight. Do you think people would want a defacto favicon in the built version?

**Comment by zeke** [comment]
> Do you think people would want a defacto favicon in the built version?

Probably not. Just clarifying your intent. :)

**Comment by wmhilton** [comment]
Oh boy! My identicon PR already is merged and on npm! :)

**Comment by wmhilton** [comment]
How seriously do we take IE support? Apparently some browser STILL doesn't accept PNG favicons. There's a `to-ico` module that can convert a PNG buffer to an ICO buffer, but it'll drag in a bunch more dependencies.

**Comment by wmhilton** [comment]
(grump grump grump) Is there not a way to turn an existing issue *into* a pull request? I feel like the discussion should move to the pull request #63 but it's annoying that it's gonna fragment the conversation thread by doing that.

**Comment by zeke** [comment]
> Is there not a way to turn an existing issue into a pull request?

Pretty sure you can still do it with https://hub.github.com/

**Comment by wmhilton** [comment]
Oh my. Googling that led to some loooong discussion threads. It sounds like the recommended practice is to separate concerns: issues for the bug / feature itself, pull-requests for the implementation details. Now I know, let's go with the flow. :)

**Comment by jdormit** [comment]
Merged in #63
